src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/PEImage.cs

```diff
@@ -113,6 +121,56 @@ private PEImage(Stream stream, bool leaveOpen, bool isVirtual)
                 for (int i = 0; i < _directories.Length; i++)
                     if (!TryRead(out _directories[i]))
                         break;
+
+            }
+        }
+
+        public void Relocate(ulong loadedImageBase)
```

Comment:
I'm not a fan of Relocate being public.  If we went the MemoryStream route, Relocate should be private and it should either be called at the end of the constructor, or we should do it on-demand when we know we have to relocate the image.